tooling/minitest/src/tests/unwinding.rs

```diff
@@ -0,0 +1,204 @@
+use crate::*;
+
+#[test]
+fn abort_in_cleanup() {
+    let mut p = ProgramBuilder::new();
+    let mut f = p.declare_function();
+
+    let cleanup = f.cleanup(|f| {
+        f.abort();
+    });
+
+    f.start_unwind(cleanup);
+    let f = p.finish_function(f);
+    let p = p.finish_program(f);
+    dump_program(p);
+    assert_abort::<BasicMem>(p, "aborted");
+}
+
+#[test]
+fn start_unwind_in_main() {
+    let mut p = ProgramBuilder::new();
+    let mut f = p.declare_function();
+
+    let cleanup = f.cleanup(|f| {
+        f.print(const_int(42));
+        f.exit(); //Call exit() instead of abort, because there is currently no way to access the standard output if the program has aborted.
+    });
+    f.start_unwind(cleanup);
+    let f = p.finish_function(f);
+    let p = p.finish_program(f);
+    dump_program(p);
+    assert_eq!(get_stdout::<BasicMem>(p).unwrap(), &["42"]);
+}
+
+#[test]
+fn unwind_recursive_func() {
+    let mut p = ProgramBuilder::new();
+
+    let f = {
+        let mut f = p.declare_function();
+        let arg = f.declare_arg::<i32>();
+        let var = f.declare_local::<i32>();
+        let ret = f.declare_ret::<i32>();
+
+        let cleanup_resume = f.cleanup_resume();
+        let cleanup_print = f.cleanup(|f| {
+            f.print(load(arg));
```

Comment:
I think it'd be fun to print both at the top of the function and in cleanup, so we can see the nesting.

Maybe make the number smaller than 10 to make the output shorter. ;)